xclim/sdba/measures.py

```diff
@@ -442,3 +442,66 @@ def _scorr(
 scorr = StatisticalPropertyMeasure(
     identifier="Scorr", aspect="spatial", compute=_scorr, allowed_groups=["group"]
 )
+
+
+def _taylordiagram(
+    sim: xr.DataArray,
+    ref: xr.DataArray,
+    dim: str = "time",
+    group: str | Grouper = "time",
+) -> xr.DataArray:
+    """Taylor diagram.
+
+    Compute the respective standard deviations of a simulation and a reference array, as well as the Pearson
+    correlation coefficient between both, all necessary parameters to plot points on a Taylor diagram.
+
+    Parameters
+    ----------
+    sim : xr.DataArray
+        data from the simulation (a time-series for each grid-point)
+    ref : xr.DataArray
+        data from the reference (observations) (a time-series for each grid-point)
+    dim : str
+        Dimension across which the correlation and standard deviation should be computed.
+    group : str
+        Compute the property and measure for each temporal groups individually.
+        Currently not implemented.
+
+
+    Returns
+    -------
+    xr.DataArray, [same as ref]
+        Standard deviations of sim, ref and correlation coefficient between both.
+    """
+    corr = xr.corr(sim, ref, dim=dim)
+
+    ref_std = ref.std(dim=dim, skipna=True, keep_attrs=True)
+    sim_std = sim.std(dim=dim, skipna=True, keep_attrs=True)
+
+    coords = {"taylor_param": ["ref_std", "sim_std", "corr"]}
+    dims = ["taylor_param"]
+
+    for coord in ref_std.coords:
+        coords[coord] = ref_std[coord]
+    for dim in ref_std.dims:
+        dims.append(dim)
+
+    out = xr.DataArray(
+        [ref_std, sim_std, corr],
+        coords=coords,
+        dims=dims,
+        attrs={
+            "correlation_type": "Pearson correlation coefficient",
+            "units": ref_std.units,
+        },
+    )
```

Comment:
I come in late, sorry.

I think you could simplify this:
```suggestion
    new_dim = xr.DataArray(["ref_std", "sim_std", "corr"], dims=("taylor_param",), name="taylor_param")

    out = xr.concat(
        [ref_std, sim_std, corr],
        new_dim,
        compat='override'  # Take common coords from `ref_std`.
    ).assign_attrs({
            "correlation_type": "Pearson correlation coefficient",
            "units": ref.units,
        }
    )
```

To support my suggestion : I think the previous code would have failed if the non-time dimensions of `sim` and `ref` were not in the same order as the `dims` list was passed explicitly in the DataArray constructor.

Reply:
Great, this is indeed much simpler. Thanks!

Reply:
I added `coords='minimal'`, as xarray didn't allow the default `coords='different'` to be passed with `compat='override'`.